econml/cate_estimator.py

```diff
@@ -36,25 +35,22 @@ def __init__(self, inference):
         """
         if inference in self._inference_options:
             inference = self._inference_options[inference]
+        self._inference = inference
         if isinstance(inference, BootstrapOptions):
             # Note that fit (and other methods) check for the presence of a _bootstrap attribute
             # to determine whether to delegate to that object or not;
             # The clones wrapped inside the BootstrapEstimator will not have that attribute since
             # it's assigned *after* creating the estimator
-            self._bootstrap = BootstrapEstimator(self, inference.n_bootstrap_samples, inference.n_jobs)
-        self._inference = inference
+            self._bootstrap = BootstrapEstimator(self, inference.n_bootstrap_samples,
+                                                 inference.n_jobs, compute_means=False)
```

Comment:
I've changed this around a bit; we still use the existing `BootstrapEstimator` as an implementation detail but in a cleaner way (I think) - we create a new wrapper whenever `fit` is called, rather than at instantiation, so there's no confusion about recursion, etc.

We can refactor this further at a later date to completely remove the `BootstrapEstimator` class, but that will require more changes to our notebooks, documentation, etc.